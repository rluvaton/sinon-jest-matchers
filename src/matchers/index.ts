export { sinonToBeCalled } from './sinonToBeCalled';
export { sinonToBeCalledTimes } from './sinonToBeCalledTimes';
export { sinonToBeCalledWith } from './sinonToBeCalledWith';
export { sinonLastCalledWith } from './sinonLastCalledWith';
export { sinonLastReturnedWith } from './sinonLastReturnedWith';
export { sinonNthCalledWith } from './sinonNthCalledWith';
export { sinonToReturn } from './sinonToReturn';
export { sinonToReturnTimes } from './sinonToReturnTimes';
export { sinonToReturnWith } from './sinonToReturnWith';
