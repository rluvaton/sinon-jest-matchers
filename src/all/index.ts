import * as matchers from '../matchers';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const jestExpect = global.expect;

if (jestExpect !== undefined) {
  jestExpect.extend(matchers);
} else {
  throw new Error(
    "Unable to find Jest's global expect. " +
      'Please check you have added sinon-jest-matchers correctly to your jest configuration. ' +
      'See https://github.com/rluvaton/sinon-jest-matchers#setup for help.',
  );
}
