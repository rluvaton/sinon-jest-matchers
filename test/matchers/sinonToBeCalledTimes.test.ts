import sinon from 'sinon';
import * as matchers from '../../src';

expect.extend(matchers);

describe('sinonToBeCalledTimes', () => {
  test('.not works only on spies or jest.fn', () => {
    const fn = function fn() {};

    expect(() => expect(fn).not.sinonToBeCalledTimes(2)).toThrowErrorMatchingSnapshot();
  });

  test('only accepts a number argument', () => {
    const fn = sinon.spy();
    fn();
    expect(fn).sinonToBeCalledTimes(1);

    [{}, [], true, 'a', new Map(), () => {}].forEach(value => {
      // @ts-expect-error we expect arg different from a number
      expect(() => expect(fn).sinonToBeCalledTimes(value)).toThrowErrorMatchingSnapshot();
    });
  });

  test('.not only accepts a number argument', () => {
    const fn = sinon.spy();
    expect(fn).not.sinonToBeCalledTimes(1);

    [{}, [], true, 'a', new Map(), () => {}].forEach(value => {
      // @ts-expect-error we expect arg different from a number
      expect(() => expect(fn).not.sinonToBeCalledTimes(value)).toThrowErrorMatchingSnapshot();
    });
  });

  test('passes if function called equal to expected times', () => {
    const fn = sinon.spy();
    fn();
    fn();

    expect(fn).sinonToBeCalledTimes(2);

    expect(() => expect(fn).not.sinonToBeCalledTimes(2)).toThrowErrorMatchingSnapshot();
  });

  test('.not passes if function called more than expected times', () => {
    const fn = sinon.spy();
    fn();
    fn();
    fn();

    expect(fn).sinonToBeCalledTimes(3);
    expect(fn).not.sinonToBeCalledTimes(2);

    expect(() => expect(fn).sinonToBeCalledTimes(2)).toThrowErrorMatchingSnapshot();
  });

  test('.not passes if function called less than expected times', () => {
    const fn = sinon.spy();
    fn();

    expect(fn).sinonToBeCalledTimes(1);
    expect(fn).not.sinonToBeCalledTimes(2);

    expect(() => expect(fn).sinonToBeCalledTimes(2)).toThrowErrorMatchingSnapshot();
  });

  test('includes the custom mock name in the error message', () => {
    const fn = jest.fn().mockName('named-mock');
    fn();

    expect(() => expect(fn).sinonToBeCalledTimes(2)).toThrowErrorMatchingSnapshot();
  });
});
