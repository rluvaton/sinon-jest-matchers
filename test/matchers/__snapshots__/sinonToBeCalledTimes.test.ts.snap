// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sinonToBeCalledTimes .not only accepts a number argument 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  object
Expected has value: [32m{}[39m"
`;

exports[`sinonToBeCalledTimes .not only accepts a number argument 2`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  array
Expected has value: [32m[][39m"
`;

exports[`sinonToBeCalledTimes .not only accepts a number argument 3`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  boolean
Expected has value: [32mtrue[39m"
`;

exports[`sinonToBeCalledTimes .not only accepts a number argument 4`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  string
Expected has value: [32m\\"a\\"[39m"
`;

exports[`sinonToBeCalledTimes .not only accepts a number argument 5`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  map
Expected has value: [32mMap {}[39m"
`;

exports[`sinonToBeCalledTimes .not only accepts a number argument 6`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  function
Expected has value: [32m[Function anonymous][39m"
`;

exports[`sinonToBeCalledTimes .not passes if function called less than expected times 1`] = `
"[2mexpect([22m[31msinon.spy()[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

Expected number of calls: [32m2[39m
Received number of calls: [31m1[39m"
`;

exports[`sinonToBeCalledTimes .not passes if function called more than expected times 1`] = `
"[2mexpect([22m[31msinon.spy()[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

Expected number of calls: [32m2[39m
Received number of calls: [31m3[39m"
`;

exports[`sinonToBeCalledTimes .not works only on spies or jest.fn 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [31mreceived[39m value must be a stub or spy function

Received has type:  function
Received has value: [31m[Function fn][39m"
`;

exports[`sinonToBeCalledTimes includes the custom mock name in the error message 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [31mreceived[39m value must be a stub or spy function

Received has type:  function
Received has value: [31m[Function mockConstructor][39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  object
Expected has value: [32m{}[39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 2`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  array
Expected has value: [32m[][39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 3`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  boolean
Expected has value: [32mtrue[39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 4`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  string
Expected has value: [32m\\"a\\"[39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 5`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  map
Expected has value: [32mMap {}[39m"
`;

exports[`sinonToBeCalledTimes only accepts a number argument 6`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

[1mMatcher error[22m: [32mexpected[39m value must be a non-negative integer

Expected has type:  function
Expected has value: [32m[Function anonymous][39m"
`;

exports[`sinonToBeCalledTimes passes if function called equal to expected times 1`] = `
"[2mexpect([22m[31msinon.spy()[39m[2m).[22mnot[2m.[22msinonToBeCalledTimes[2m([22m[32mexpected[39m[2m)[22m

Expected number of calls: not [32m2[39m"
`;
